xonsh/completer.py

```diff
@@ -78,16 +78,54 @@ def _normpath(p):
 
     return p
 
-class Completer(object):
-    """This provides a list of optional completions for the xonsh shell."""
+class KnownCommands(object):
```

Comment:
Also, it might make sense to move this class out of `completer.py` and into, e.g., `tools.py`, since it is being used multiple places (shell, completer, pygments lexer all use it).  But that's maybe a smaller issue.